typescript/packages/mechanisms/evm/src/exact/server/scheme.ts

```diff
@@ -177,6 +177,9 @@ export class ExactEvmScheme implements SchemeNetworkServer {
    */
   private getDefaultAsset(network: Network): { address: string; name: string; version: string } {
     // Map of network to USDC info including EIP-712 domain parameters
+    // Each network has the right to determine it's own default stablecoin that can be expressed as a USD string by calling servers
+    // NOTE: Currently only EIP-3009 supporting stablecoins can be used with this scheme
+    // Generic ERC20 support via EIP-2612/permit2 is planned, but not yet implemented.
     const usdcInfo: Record<string, { address: string; name: string; version: string }> = {
       "eip155:8453": {
         address: "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913",
```

Comment:
Suggest adding an explicit decimals field here and remove the `getAssetDecimals` function that might be too USDC biased
```suggestion
    const usdcInfo: Record<string, { address: string; name: string; version: string, decimals: string }> = {
      "eip155:8453": {
        address: "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913",
        name: "USD Coin",
        version: "2",
        decimals: "6",
      },
```

Reply:
Good catch